src/shogun/clustering/Hierarchical.cpp

```diff
@@ -212,8 +217,83 @@ SGMatrix<int32_t> CHierarchical::get_cluster_pairs()
 	return SGMatrix<int32_t>(pairs,2,merges, false);
 }
 
-
 void CHierarchical::store_model_features()
 {
-	/* TODO. Currently does nothing since apply methods are not implemented. */
+	CDenseFeatures<float64_t>* rhs = 
+		(distance->get_rhs())->as<CDenseFeatures<float64_t>>();
```

Comment:
I don't think you need the parenthesis here.
Also, are we really sure that the features are float64 bit?

Reply:
Will remove the parenthesis.
I checked the [KmeansBase](https://github.com/shogun-toolbox/shogun/blob/58b914a6b326cd5f799727e39737bc4b4b11b286/src/shogun/clustering/KMeansBase.cpp#L249) file and used float64 bit. I felt that it made sense to make it float64 bit. What other type do you suggest?

Reply:
KMeans is another algorithm. I just checked `Hierarchical.cpp` and it does **not** fix the features to be 64 bit float, but rather just calles the `distance` method of `CDistance`. So we will need to think a bit more here in order to not make the algorithm only work for those 64bit dense feautres. What about 32 bit? What about sparse features?` What about string features? But let's address the other points first, this one will take some time and effort

Reply:
Okay. I get your point. Thanks!